src/pages/profile/ProfilePage.js

```
@@ -13,11 +14,13 @@ import { LabelledValue } from 'src/components';
 
 export class ProfilePage extends Component {
   updateProfile = (values) => {
+    const { showAlert } = this.props;
     const { username } = this.props.currentUser;
-    return this.props.updateUser(username, { first_name: values.firstName, last_name: values.lastName })
-      .catch(() => {
-        showAlert({ type: 'error', message: 'Unable to update profile' });
-      });
+    const data = { first_name: values.firstName, last_name: values.lastName };
+
+    return this.props.updateUser(username, data)
+      .then(this.props.getCurrentUser)
```

i considered putting two `catch` block here (one for update and one for get) but seemed overkill. let me know if you think two `catch`es make sense.

---

Might not be a bad idea to catch the `updateUser` action and do the `showAlert`, and maybe just swallow any `getCurrentUser` errors? The update succeeded in that case so it seems bad to say the update failed, not sure we really have to worry about that case that much and can just swallow/ignore...

---

the messaging for the 2nd catch will be different like unable to reload user or such. it's because, unless we communicate that if you comes back to this page (without refresh), he'll still see old value thinking the update is failed.

---

Yeah I understand, not sure the messaging will be helpful there "We updated your profile but failed to update it in the client, please refresh" just sounds bizarre/weird. This scenario is so unlikely that I'm not worried if the UI looks out of sync, I think, and just ignore it.

---

Or just put a `window.location.reload()` in the 2nd catch :P ;)

---

lol. ok. i think list we can do move the catch to update. so at least get user error will surface to console.